### PR TITLE
Adding binaries, sensors, and battery - Separating devices

### PR DESCRIPTION
**Pull request by doonce**
Repost of [PR#41](https://github.com/ymerj/HE-HA-control/pull/41)

I did my best to compare the binary sensors within HA and match them up with their Hubitat counterparts. I had to take a new direction in translating devices since there is a "lock" device and "lock" binary_sensor, so it will now look based on domain as well as device_class. Binary sensor classes are [here](https://www.home-assistant.io/integrations/binary_sensor/), sensor classes are [here](https://www.home-assistant.io/integrations/sensor#device-class), and it will assume other things are devices.

There are a few HA binary sensors that **should** work in Hubitat, but there isn't a Generic Component available including gas sensor, sound sensor, tamper sensor, light sensor (also conflicts with light device, another reason I separated), etc. However, a lot of these can be hacked into use by just setting them as motion sensors or another Generic Binary Component, but I didn't want to do that here.

Everything I added has been tested using template devices in HA.

### CODE REVIEW

**Comment by doonce** [comment]
I do still think removing whitespace is important, but I have left it out of this PR for now. Also, as I mentioned in the HE forums, it is possible to add some additional missing generic component devices to add support of HA devices in HE like I have done here with Battery.

File: HA parent.groovy [hunk 1]
```diff
@@ -59,6 +59,7 @@
 * 0.1.36 2021-12-14 Yves Mercier       Improved event type
 * 0.1.37 2021-12-26 gabriel_kpk        Added support for Climate domain
 * 0.1.38 2021-12-29                    Improved Climate support, Code cleanup, Minor decription fixes
+* 0.1.38 2022-01-19 BrenenP            Added support for additional sensors
```
Comment (tomwpublic) [inline HA parent.groovy]: Please update to version 0.1.39.

File: genericComponentBattery.groovy [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+
+-------------------------------------------
+
+Change history:
+
+0.1.13- @tomw - initial version
```
Comment (tomwpublic) [inline genericComponentBattery.groovy]: Please update to version 0.1.39 and attribute to BrenenP.

File: genericComponentBattery.groovy [hunk 3]
```diff
@@ -0,0 +1,61 @@
+/*
+
+Copyright 2021
```
Comment (tomwpublic) [inline genericComponentBattery.groovy:3]: Please add this file as an optional component in packageManifest.json. 'id' is just a random GUID.  Also update to version 0.1.39 in packageManifest.json

File: HA parent.groovy [hunk 4]
```diff
@@ -292,30 +302,51 @@ def parse(String description) {
...
-            window: [type: "Generic Component Contact Sensor",          event: [[name: "contact", value: newVals[0] == "on" ? "open":"closed", descriptionText:"${friendly} is ${newVals[0]}"]]],
-            device_tracker: [type: "Generic Component Presence Sensor", event: [[name: "presence", value: newVals[0] == "home" ? "present":"not present", descriptionText:"${friendly} is ${newVals[0]}"]], namespace: "community"],
-            cover: [type: "Generic Component Garage Door Control",      event: [[name: "door", value: newVals[0] ?: "unknown", type: origin, descriptionText:"${friendly} was turn ${newVals[0]} [${origin}]"]], namespace: "community"],
-            lock: [type: "Generic Component Lock",                      event: [[name: "lock", value: newVals[0] ?: "unknown", type: origin, descriptionText:"${friendly} was turn ${newVals[0]} [${origin}]"]]],
+        ]
+
+    return mapping[device_class]
+}
+
+def translateDevices(device_class, newVals, friendly, origin)
```
Comment (tomwpublic) [inline HA parent.groovy]: Rename device_class parameter to domain for clarity.

**Review by tomwpublic — CHANGES_REQUESTED** [review]

**Pull request merged**